crates/sncast/src/lib.rs

```diff
@@ -168,11 +168,12 @@ pub async fn get_account<'a>(
     keystore: Option<Utf8PathBuf>,
 ) -> Result<SingleOwnerAccount<&'a JsonRpcClient<HttpTransport>, LocalWallet>> {
     let chain_id = get_chain_id(provider).await?;
-    let account = if let Some(keystore) = keystore {
+    let mut account = if let Some(keystore) = keystore {
         get_account_from_keystore(provider, chain_id, &keystore, account)?
     } else {
         get_account_from_accounts_file(account, accounts_file, provider, chain_id)?
     };
+    let account = account.set_block_id(get_block_id("pending")?).clone();
```

Comment:
isn't it possible to just 
```
account.set_block_id(...)
```
without reassignment?

Reply:
![Zrzut ekranu 2024-01-19 o 11 25 39](https://github.com/foundry-rs/starknet-foundry/assets/15279784/822ad574-54f8-48f2-ac8a-21465a9e4ffa)

Reply:
This just means you need to return an account, and not a refference to it. you can derefference or clone, should be fine.

Reply:
I can't derefference because Account does not implement the trait `Copy`

Reply:
I see you've managed to do this :+1: